src/shell.js

```diff
@@ -137,27 +137,51 @@ var quit_ = (status, toThrow) => {
   throw toThrow;
 };
 
-#if SHARED_MEMORY && !MODULARIZE
-// In MODULARIZE mode _scriptName needs to be captured already at the very top of the page immediately when the page is parsed, so it is generated there
-// before the page load. In non-MODULARIZE modes generate it here.
-var _scriptName = (typeof document != 'undefined') ? document.currentScript?.src : undefined;
-
-#if ENVIRONMENT_MAY_BE_NODE
-if (ENVIRONMENT_IS_NODE) {
 #if EXPORT_ES6
-  _scriptName = typeof __filename != 'undefined' ? __filename : import.meta.url
+var _scriptName = import.meta.url;
```

Comment:
It looks like the declaration of this variable was complete elided previous unless `#if SHARED_MEMORY && !MODULARIZE`.   Can we not continue to default to not defining it at all?

Reply:
The idea was to always ensure that `_scriptName` is defined, so that we can do our `scriptDirectory` logic on top of that (which is just `new URL('.', _scriptName).href` on the web). Note that `_scriptName` was also previously defined in `MODULARIZE` mode, but outside the wrapper function.

In `EXPORT_ES6` mode it was also previously defined outside the wrapper function, but I moved it to here since storing `import.meta.url` there isn't necessary (afaik).